src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java

```diff
@@ -0,0 +1,260 @@
+package org.broadinstitute.hellbender.tools.walkers.vqsr;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.text.XReadLines;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+/*
+ * Tranche in VQSR.
+ * Package-private because it's not usable outside.
+ */
+final class Tranche {
+    public final double targetTruthSensitivity;
+    public final double minVQSLod;  //minimum value of VQSLOD in this tranche
+    public final double knownTiTv;  //titv value of known sites in this tranche
+    public final double novelTiTv;  //titv value of novel sites in this tranche
+    public final int numKnown;      //number of known sites in this tranche
+    public final int numNovel;      //number of novel sites in this tranche
+    public final String name;       //Name of the tranche
+    public final VariantRecalibratorArgumentCollection.Mode model;    //this is a SNP VQSR tranche or Indel tranche?
+
+    private final int accessibleTruthSites;
+    private final int callsAtTruthSites;
+
+    public Tranche(double targetTruthSensitivity, double minVQSLod, int numKnown, double knownTiTv, int numNovel, double novelTiTv, int accessibleTruthSites, int callsAtTruthSites, VariantRecalibratorArgumentCollection.Mode model) {
+        this(targetTruthSensitivity, minVQSLod, numKnown, knownTiTv, numNovel, novelTiTv, accessibleTruthSites, callsAtTruthSites, model, "anonymous");
+    }
+
+    public Tranche(double targetTruthSensitivity, double minVQSLod, int numKnown, double knownTiTv, int numNovel, double novelTiTv, int accessibleTruthSites, int callsAtTruthSites, VariantRecalibratorArgumentCollection.Mode model, String name) {
+        if ( targetTruthSensitivity < 0.0 || targetTruthSensitivity > 100.0) {
+            throw new UserException("Target FDR is unreasonable " + targetTruthSensitivity);
+        }
+
+        if ( numKnown < 0 || numNovel < 0) {
+            throw new GATKException("Invalid tranche - no. variants is < 0 : known " + numKnown + " novel " + numNovel);
```

Comment:
Are you sure that an invalid `targetTruthSensitivity` is the user's fault, while an invalid number of known/novel variants is our fault?